schema.graphql

```diff
@@ -120,3 +290,262 @@ type GenesisFile @entity {
   id: ID!
   raw: String!
 }
+
+type SupplierStakeMsg @entity {
+  id: ID!
+  signer: Account!
+  supplier: Supplier!
+  owner: Account!
+  stake: Coin!
+  block: Block!
+  transaction: Transaction!
+}
+
+type SupplierStakeMsgService @entity {
+  id: ID!
+  supplierStakeMsg: SupplierStakeMsg!
+  service: Service!
+  revShare: [SupplierRevShare]!
+  endpoints: [SupplierEndpoint]!
+}
+
+type SupplierUnstakeMsg @entity {
+  id: ID!
+  signer: Account!
+  supplier: Supplier!
+  block: Block!
+  transaction: Transaction!
+}
+
+type Supplier @entity {
+  id: ID!
+  operator: Account!
+  owner: Account!
+  stake: Coin!
+  stakeMsgs: [SupplierStakeMsg] @derivedFrom(field: "supplier")
+  unstakeMsgs: [SupplierUnstakeMsg] @derivedFrom(field: "supplier")
+  unstakingBeginEvents: [SupplierUnbondingBeginEvent] @derivedFrom(field: "supplier")
+  unstakingEndEvents: [SupplierUnbondingEndEvent] @derivedFrom(field: "supplier")
+  # 0 = Staked, 1 = Unstaking, 2 = Unstaked
+  status: Int!
+  # Block where MsgUnstakeSupplier was received
+  unstakingStartBlock: Block
+  # Height of the block where the supplier is going to be unstaked/unbonded.
+  # This is received with the EventSupplierUnbondingBegin event.
+  unstakingHeight: BigInt
+  # Block where the supplier was unstaked/unbonded.
+  # This is received with the EventSupplierUnbondingEnd event.
+  unstakedAtBlock: Block
+}
+
+type ConfigOption @jsonField {
+  key: Int!
+  value: String!
+}
+
+type SupplierEndpoint @jsonField {
+  url: String!
+  rpcType: Int!
+  configs: [ConfigOption]!
+}
+
+type SupplierRevShare @jsonField {
+  address: String!
+  revSharePercentage: Float!
+}
+
+type SupplierService @entity {
```

Comment:
Oh, it looks like this could be repurposed to address the comment above regarding `SupplierServiceConfig`...